app/models/restroom.rb

```diff
@@ -18,13 +18,14 @@ class Restroom < ApplicationRecord
   ignoring: :accents
 
   validates :name, :street, :city, :state, presence: true
+  validates :street, uniqueness: {scope: [:city, :state], message: "is already registered"}
 
   geocoded_by :full_address
-  after_validation :perform_geocoding
+  before_validation :perform_geocoding, if: ->(obj){ obj.full_address.present? }
+  before_validation :reverse_geocode, if: ->(obj){ obj.full_address.present? }
```

Comment:
I think you can DRY this and make it a little simpler by using a method here.
https://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless

```ruby
before_validation :perform_geocoding, if: :require_geocoding?
```
```ruby
def require_geocoding?
  full_address.present?
end
```

Also, `full_address` should be reimplemented, since right now it always returns a string (so this condition is always `true`. 

Instead of:
```ruby
def full_address
  "#{street}, #{city}, #{state}, #{country}"
end
```

```ruby
def full_address
  [street, city, state, country].compact.join(",")
end
```
(I think this is better than showing empty values, but "123 Main Street, USA" doesn't really make sense either.)